### DISCUSSION TITLE
Static assets in Laravel not working with Serverless-lift

### DISCUSSION THREAD

**Discussion by mikevang** [brefphp/bref#1159] [Support] [answered]
I've followed the instructions for serving static assets from s3 as outlined in [https://bref.sh/docs/frameworks/laravel.html#assets](https://bref.sh/docs/frameworks/laravel.html#assets).

In my blade file I'm trying to load an image from /public/images, and using the asset helper as described in the documentation: 
`<img src="{{ asset('images/loading.gif') }}" class="loading-overlay-img"/>`

This works fine locally, however when I deploy to aws and load the page via cloudfront, it shows a broken image. Then looking at the served code, image is pointing to `https://<prefix>.execute-api.us-west-2.amazonaws.com/images/loading.gif` when it should be pointing to `https://<prefix>.cloudfront.net/images/loading.gif`

I have the following in my serverless.yml
```
constructs:
  website:
    type: server-side-website
    assets:
      '/js/*': public/js
      '/css/*': public/css
      '/images/*': public/images
      '/favicon.ico': public/favicon.ico
      '/robots.txt': public/robots.txt
```

What am I doing wrong here?

**Comment by t-richard** [comment, marked as answer]
Hi

I think you should make sure your `TrustProxies` middleware is configured with `Request::HEADER_X_FORWARDED_FOR | Request::HEADER_X_FORWARDED_HOST | Request::HEADER_X_FORWARDED_PORT | Request::HEADER_X_FORWARDED_PROTO` in `$headers` as showed at https://laravel.com/docs/8.x/requests#configuring-trusted-proxies

The important part here is the `HEADER_X_FORWARDED_HOST`, it makes sure laravel will use the value in the `x-forwarded-host` to generate URLs and this is automatically set by Lift's construct.

Hope it helps. If it does, I think a PR to the docs would be welcome :slightly_smiling_face:

**Reply by mikevang** [reply]
My TrustProxies is already configured with `HEADER_X_FORWARDED_HOST` 
```
protected $headers =
        Request::HEADER_X_FORWARDED_FOR |
        Request::HEADER_X_FORWARDED_HOST |
        Request::HEADER_X_FORWARDED_PORT |
        Request::HEADER_X_FORWARDED_PROTO |
        Request::HEADER_X_FORWARDED_AWS_ELB;
```

**Reply by t-richard** [reply]
Maybe you could try to trust all proxies then with `protected $proxies = '*';` like shown here: https://laravel.com/docs/8.x/requests#trusting-all-proxies

**Reply by mikevang** [reply]
> Maybe you could try to trust all proxies then with `protected $proxies = '*';` like shown here: https://laravel.com/docs/8.x/requests#trusting-all-proxies

That seems to have done the trick! Thanks.

**Reply by t-richard** [reply]
FYI, the Laravel bridge is supposed to handle this automatically but it's broken on new installs since Laravel 8

See the details here: https://github.com/brefphp/laravel-bridge/issues/48#issuecomment-1030161470

**Reply by mnapoli** [reply]
Until we find a way to automatically configure it, it's now documented on bref.sh: https://bref.sh/docs/frameworks/laravel.html#trusted-proxies

**Reply by cirolosapio** [reply]
i'm facing the same issue
is the documented part removed from bref docs?

**Reply by cirolosapio** [reply]
oh i missed the cloudfront url
it works 👍
